src/WooCommerce.php

```diff
@@ -463,6 +463,12 @@ public static function woocommerce_get_price_html($price, $product) {
         }
         $price = wc_price(wc_get_price_to_display($product)) . $product->get_price_suffix();
       }
+      $current_filter = current_filter();
+      $priority = has_filter($current_filter, __CLASS__ . '::woocommerce_get_price_html');
+      // Remove and re-add filter to avoid callback loop.
+      remove_filter($current_filter, __CLASS__ . '::woocommerce_get_price_html');
```

Comment:
https://developer.wordpress.org/reference/functions/remove_filter/ states
> To remove a hook, the $function_to_remove and $priority arguments must match when the hook was added. This goes for both filters and actions. No warning will be given on removal failure.

The default value is 10, but this will fail if the priority was changed.
```suggestion
      remove_filter($current_filter, __CLASS__ . '::woocommerce_get_price_html', $priority);
```

I also wonder whether the callback function name isn't the same as `__FUNCTION__` in this context, but that's a very minor detail.

Similarly minor, the inline comment should be above the block of lines, not in between, and should explain what is being done; e.g.:
```php
// Invoke regular price_html filters (excluding this one).
```

Reply:
Thanks for catching! Overseen this as well.

Reply:
Added a fixup using `__METHOD__` as this returns the full path while `__FUNCTION__` returns the name only.